### PR TITLE
Expand docs for recv

### LINKED ISSUE #26497: Document what happens when Receiver::recv() is called on a closed channel when there are queued messages
http://doc.rust-lang.org/std/sync/mpsc/struct.Receiver.html#method.recv states that if the sender has disconnected, an `Err` result is returned. This is only partially true - `recv()` will first run on every message that was queued before the Sender disconnected, then return err.

Example of this: https://play.rust-lang.org/?gist=f9d9741f1e5ae5dca3c1&version=stable

I think the documentation deserves at least a mention of this behavior - as it is kind of ambiguous whether the receiver will return Err even if there are messages queued or not just reading the current documentation.

### PR DESCRIPTION
**Pull request by steveklabnik**
Add an example, plus some text that covers the buffering nature of
channels.

Fixes #26497

### CODE REVIEW

File: src/libstd/sync/mpsc/mod.rs [hunk 1]
```diff
@@ -768,6 +768,48 @@ impl<T> Receiver<T> {
...
+    ///  
+    /// let (send, recv) = mpsc::channel();
+    /// let handle = thread::spawn(move || {
+    ///     send.send(1u8).unwrap();
+    ///     send.send(2).unwrap();
+    ///     send.send(3).unwrap();
+    ///     drop(send);
+    /// });
+    ///
+    /// // `send` is now dropped, but the messages were still buffered
```
Comment (huonw) [inline src/libstd/sync/mpsc/mod.rs]: maybe this comment should be after the `join`?

**Comment by huonw** [comment]
tidy failed with some trailing whitespace

File: src/libstd/sync/mpsc/mod.rs (hunk 1 above)
Comment (steveklabnik) [inline src/libstd/sync/mpsc/mod.rs]: I put comments describing a line of code before that line, but i see what you're saying with the phrasing

Comment (huonw) [inline src/libstd/sync/mpsc/mod.rs]: Yeah, I'm thinking temporarily, the "now" only applies after this statement.

**Comment by steveklabnik** [comment]
Tidy fixed and I just removed the comment, I think it stands enough on its own, but maybe not?

**Comment by huonw** [comment]
Hm, it's not entirely obvious. A possibility is something like "wait for the thread to finish so that the sender is dropped"?

r=me with a comment like that or not, whatever you feel is best.

**Comment by steveklabnik** [comment]
i like it, added

**Comment by BigBoneDaddy** [comment]
Much much much clear now!

**Pull request merged**